src/relay/backend/build_module.cc

```diff
@@ -0,0 +1,669 @@
+    }
+    if (cfg.pass_enabled("EliminateCommonSubexpr")) {
+      auto fskip = PackedFunc([](TVMArgs args, TVMRetValue* rv) {
+        Expr expr = args[0];
+        if (expr.as<CallNode>()) {
+          auto call_node = expr.as<CallNode>();
+          auto op_node = call_node->op.as<OpNode>();
+          if (op_node->name == "cast") {
+            auto attrs = call_node->attrs.as<CastAttrs>();
+            if (attrs->dtype == HalideIR::Int(32)) {
+              return true;
+            }
+          }
+        }
+        return false;
+      });
+      func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+      func = _CallPacked("relay._ir_pass.eliminate_common_subexpr", func, fskip);
+    }
+    if (cfg.pass_enabled("CombineParallelConv2D")) {
+      func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+      func = _CallPacked("relay._ir_pass.CombineParallelConv2D", func);
+    }
+    if (cfg.pass_enabled("FoldConstant")) {
+      func = _CallPacked("relay._ir_pass.FoldConstant", func);
+    }
+    if (cfg.pass_enabled("FoldScaleAxis")) {
+      func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+      func = _CallPacked("relay._ir_pass.backward_fold_scale_axis", func);
+      func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+      func = _CallPacked("relay._ir_pass.forward_fold_scale_axis", func);
+      func = _CallPacked("relay._ir_pass.FoldConstant", func);
+    }
+    if (cfg.pass_enabled("CanonicalizeOps")) {
+      func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+      func = _CallPacked("relay._ir_pass.canonicalize_ops", func);
+    }
+    if (cfg.pass_enabled("AlterOpLayout")) {
+      if (targets.size() == 1) {
+        func = _CallPacked("relay._ir_pass.infer_type", func, nullptr);
+        func = _CallPacked("relay._ir_pass.AlterOpLayout", func);
+      } else {
+        LOG(WARNING) << "AlterOpLayout pass is not enabled for heterogeneous"
+                  << " execution yet.";
+      }
+    }
+    if (cfg.pass_enabled("FoldConstant")) {
+      func = _CallPacked("relay._ir_pass.FoldConstant", func);
+    }
+    return func;
+  }
+
+  Map<HalideIR::Expr, HalideIR::Expr> _UpdateHeterogeneousInputs(
+    const std::unordered_map<std::string, std::string>& targets,
+    const RelayBuildConfig& cfg) {
+    Map<HalideIR::Expr, HalideIR::Expr> device_target;
+    std::unordered_map<int64_t, std::string> tmp_map;
+    auto fallback_idx = ContextMap::Str2Mask(cfg.fall_back_device);
+
+    for (const auto& kv : targets) {
+      tmp_map[ContextMap::Str2Mask(kv.first)] = kv.second;
+    }
+    if (tmp_map.count(fallback_idx) == 0) {
+      tmp_map[fallback_idx] = cfg.fall_back_device;
+    }
+    for (const auto& kv : tmp_map) {
+      device_target.Set(
+        ir::IntImm::make(HalideIR::Int(64), kv.first),
+        ir::StringImm::make(kv.second));
+    }
+    return device_target;
+  }
+
+  Function _RunDeviceAnnotationPass(
+    Function func,
```

Comment:
```suggestion
    const Function& func,
```

Reply:
As we are changing func during running pass, const reference are not be able to use here. Same to other place.